src/Interpreters/ExternalLoaderXMLConfigRepository.cpp

```diff
@@ -90,6 +90,9 @@ Poco::AutoPtr<Poco::Util::AbstractConfiguration> ExternalLoaderXMLConfigReposito
     ConfigProcessor config_processor{config_file};
     ConfigProcessor::LoadedConfig preprocessed = config_processor.loadConfig();
     config_processor.savePreprocessedConfig(preprocessed, app_path);
+#if USE_SSL
+    config_processor.decryptEncryptedElements(preprocessed, /* load_encryption_codecs */ false, /* decrypt_encrypted_values */ true);
```

Comment:
Similar comment as in UsersConfigAccessStorage.cpp ... it is unclear why we don't want/need to load encryption codecs. What if they are not loaded yet?

Reply:
> Similar comment as in UsersConfigAccessStorage.cpp ... it is unclear why we don't want/need to load encryption codecs. What if they are not loaded yet?

Encryption codecs are not supposed to be defined in users.xml, as well as any other config, so they are not loaded here. There is a limitation, as I mentioned before: we have only 1 attempt to load them, because of the ```CompressionCodecEncrypted::Configuration::loadImpl``` design: any new unsuccessful attempt resets good previously loaded codecs.

Reply:
> Maybe you suggest changing load() method logic? If the keys are not found, leave the previous ones as they are, not reset them?

Yes, that would be an option.